app/src/main/java/org/schabi/newpipe/fragments/detail/DescriptionFragment.java

```diff
@@ -90,4 +97,98 @@ private void setupDescription(final TextView descriptionTextView) {
                 break;
         }
     }
+
+    private void setupMetadata(final LayoutInflater inflater,
+                               final LinearLayout layout) {
+        addMetadataItem(inflater, layout, false,
+                R.string.metadata_category, streamInfo.getCategory());
+
+        addTagsMetadataItem(inflater, layout);
+
+        addMetadataItem(inflater, layout, false,
+                R.string.metadata_licence, streamInfo.getLicence());
+
+        addPrivacyMetadataItem(inflater, layout);
+
+        if (streamInfo.getAgeLimit() != NO_AGE_LIMIT) {
+            addMetadataItem(inflater, layout, false,
+                    R.string.metadata_age_limit, String.valueOf(streamInfo.getAgeLimit()));
+        }
+
+        if (streamInfo.getLanguageInfo() != null) {
+            addMetadataItem(inflater, layout, false,
+                    R.string.metadata_language, streamInfo.getLanguageInfo().getDisplayLanguage());
+        }
+
+        addMetadataItem(inflater, layout, true,
+                R.string.metadata_support, streamInfo.getSupportInfo());
+        addMetadataItem(inflater, layout, true,
+                R.string.metadata_host, streamInfo.getHost());
+        addMetadataItem(inflater, layout, true,
+                R.string.metadata_thumbnail_url, streamInfo.getThumbnailUrl());
+    }
+
+    private void addMetadataItem(final LayoutInflater inflater,
+                                 final LinearLayout layout,
+                                 final boolean linkifyContent,
+                                 @StringRes final int type,
+                                 @Nullable final String content) {
+        if (isBlank(content)) {
+            return;
+        }
+
+        final ItemMetadataBinding binding = ItemMetadataBinding.inflate(inflater, layout, false);
+        binding.metadataTypeView.setText(type);
+
+        if (linkifyContent) {
+            TextLinkifier.createLinksFromPlainText(layout.getContext(), content,
+                    binding.metadataContentView);
+        } else {
+            binding.metadataContentView.setText(content);
+        }
+
+        layout.addView(binding.getRoot());
+    }
+
+    private void addTagsMetadataItem(final LayoutInflater inflater, final LinearLayout layout) {
+        if (streamInfo.getTags() != null && !streamInfo.getTags().isEmpty()) {
+            final StringBuilder tags = new StringBuilder();
+            for (int i = 0; i < streamInfo.getTags().size(); ++i) {
+                if (i != 0) {
+                    tags.append(", ");
```

Comment:
I don't know much about localization, but I think the enumaration comma in Chineese (Mandarin) is `、`.  I don't know about other languages. I am not sure what to do here, because when using YouTube, the tags might be English, but the locale is set to Chineese.

Reply:
Oh, right. Should I create a new translatable string?

Reply:
Using chips like @wb9688 suggests would fix this issue, but it might indeed look bad if there are a lot of tags. @Stypox maybe you can give chips a try and post a screenshot so we can see how it would look like? In case they do turn out to look bad, maybe we could show part of them by default and then add some method to show all of them.

Reply:
Ok, I will try